View/DataContexts/SongListDataContext.cs

```diff
@@ -46,7 +53,20 @@ public void UpdateSongInfoList()
                     break;
                 case ScreenNames.ArtistScreen:
                     songlist = SongController.Create(new DatabaseContext()).GetList()
-                        .Where(s => s.Artist == ArtistController.Create(new DatabaseContext()).GetItem(UserController.CurrentUser.ID).ID).ToList();
+                        .Where(s => s.Artist.Equals(
+                            artistController.GetArtistIdFromUserId(UserController.CurrentUser.ID))).ToList();
+                    break;
+                case ScreenNames.SongListScreen:
+                    if ((bool)DataContext.Instance.IsAdmin)
+                    {
+                        songlist = SongController.Create(new DatabaseContext()).SearchSongsOnString(Instance.SongListSearchTerms.ToList());
+                    }
+                    else if ((bool) DataContext.Instance.IsArtist)
+                    {
+                        songlist = SongController.Create(new DatabaseContext()).SearchSongsOnString(Instance.SongListSearchTerms.ToList())
+                            .Where(s => s.Artist == artistController.GetItem((int) artistController
+                                .GetArtistIdFromUserId(UserController.CurrentUser.ID)).ID).ToList();
```

Comment:
dit is dubbelop. je vraagt eerst het id van de artiest op, daar haal je de bijbehorende artiest bij op en daar pak je dan weer het id van

Reply:
has been resolved